scripts/csv-to-json/index.ts

```diff
@@ -22,49 +30,80 @@ function csvToType(rowStrings: string[]): any {
   const firstRowString = rowStrings.shift()
   if (!firstRowString) throw new Error("No first row")
   const firstRow = firstRowString.split(',')
-  const res: any = {}
-  rowStrings.forEach(rowString => {
-    const row = rowString.split(',')
-    row.forEach((item, i) => {
-      const heading = firstRow[i]
-      if (!heading) throw new Error("No heading")
-      const keystring = row[0]
-      if (!keystring) throw new Error("No keystring in item")
-      if (i === 0) {
-        if (heading !== "keystring") throw new Error ("Expected 'keystring' heading")
-        const current = res[keystring]
-        if (!current) {
-          res[keystring] = [{}] as any[]
-        } else {
-          res[keystring] = current.concat([{}])
-        }
-      } else {
-        const objectArray: any[] = res[keystring]
-        if (!objectArray) throw new Error("No object array")
-        if (objectArray.length === 0) objectArray.push({})
-        let object = objectArray[objectArray.length - 1]
-        if (object === undefined) throw new Error("No object")
-        object[heading] = item
-        objectArray[objectArray.length - 1] = object
-        res[keystring] = objectArray
-      }
-    })
-  })
+  const type = firstRow[0]
+  let res: any = {}
+  switch (type) {
+    case undefined:
+      throw new Error("No type")
+
+    case "keystring":
+      rowStrings.forEach(rowString => {
+        const row = rowString.split(',')
+        row.forEach((item, i) => {
+          const heading = firstRow[i]
+          if (!heading) throw new Error("No heading")
+          const keystring = row[0]
+          if (!keystring) throw new Error("No keystring in item")
+          if (i === 0) {
```

Comment:
This could be done outside of the forEach call, then do the forEach on a slice from 1 to the end, or simply skip i===0 at the top